src/sys/select.rs

```diff
@@ -24,37 +27,40 @@ fn assert_fd_valid(fd: RawFd) {
     );
 }
 
-impl FdSet {
+impl<'fd> FdSet<'fd> {
     /// Create an empty `FdSet`
-    pub fn new() -> FdSet {
+    pub fn new() -> FdSet<'fd> {
         let mut fdset = mem::MaybeUninit::uninit();
         unsafe {
             libc::FD_ZERO(fdset.as_mut_ptr());
-            FdSet(fdset.assume_init())
+            Self {
+                set: fdset.assume_init(),
+                _fd: std::marker::PhantomData,
+            }
         }
     }
 
     /// Add a file descriptor to an `FdSet`
-    pub fn insert(&mut self, fd: RawFd) {
-        assert_fd_valid(fd);
-        unsafe { libc::FD_SET(fd, &mut self.0) };
+    pub fn insert<Fd: AsFd>(&mut self, fd: &'fd Fd) {
```

Comment:
For `FdSet::insert()` and `FdSet::contains()`, taking  `AsFd` by references to make them safer.